packages/sirius-web/frontend/sirius-web-application/src/views/project-settings/useViewableTabContribution.tsx

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { gql, useQuery } from '@apollo/client';
+import { useData, useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import { useEffect } from 'react';
+import {
+  GQLGetProjectData,
+  GQLGetProjectVariables,
+  GQLProjectSettingsCapabilities,
+  ProjectSettingTabContribution,
+} from './ProjectSettingsView.types';
+import { projectSettingsTabExtensionPoint } from './ProjectSettingsViewExtensionPoints';
+import { UseViewableTabContributionValue as UseProjectSettingsValue } from './useViewableTabContribution.types';
+
+const getProjectQuery = gql`
+  query getProject($projectId: ID!, $tabIds: [ID!]!) {
+    viewer {
+      project(projectId: $projectId) {
+        id
+        name
+        capabilities {
+          settings {
+            canView
+            tabs(tabIds: $tabIds) {
+              tabId
+              canView
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const useProjectSettings = (projectId: string): UseProjectSettingsValue => {
+  const { data: projectSettingsTabContributions } = useData(projectSettingsTabExtensionPoint);
+  const { loading, data, error } = useQuery<GQLGetProjectData, GQLGetProjectVariables>(getProjectQuery, {
+    variables: {
+      projectId,
+      tabIds: projectSettingsTabContributions.map((tab) => tab.id),
+    },
+  });
+
+  const { addErrorMessage } = useMultiToast();
+  useEffect(() => {
+    if (error) {
+      addErrorMessage(error.message);
+    }
+  }, [error]);
+
+  const settings: GQLProjectSettingsCapabilities | null = data?.viewer.project.capabilities.settings;
+
+  const viewableTabContributions: ProjectSettingTabContribution[] = projectSettingsTabContributions.filter(
+    (contribution) => settings?.tabs.find((tab) => tab.tabId === contribution.id && tab.canView)
+  );
+
+  return {
+    loading,
+    project: data?.viewer.project,
+    viewableTabContributions,
```

Comment:
Why return two values with one containing the other and why mix the query to the server with the extension point? Things evolve constantly and we don't want to create coupling between concepts. Just create the hook for the query to the backend.